coreax/score_matching.py

```diff
@@ -120,61 +111,45 @@ class SlicedScoreMatching(ScoreMatching):
     :param gamma: Geometric progression ratio. Defaults to 0.95.
     """
 
-    def _tree_flatten(self):
-        """
-        Flatten a pytree.
-
-        Define arrays & dynamic values (children) and auxiliary data (static values).
-        A method to flatten the pytree needs to be specified to enable jit decoration
-        of methods inside this class.
-        """
-        children = (self.random_key,)
-        aux_data = {
-            "random_generator": self.random_generator,
-            "noise_conditioning": self.noise_conditioning,
-            "use_analytic": self.use_analytic,
-            "num_random_vectors": self.num_random_vectors,
-            "learning_rate": self.learning_rate,
-            "num_epochs": self.num_epochs,
-            "batch_size": self.batch_size,
-            "hidden_dims": self.hidden_dims,
-            "optimiser": self.optimiser,
-            "num_noise_models": self.num_noise_models,
-            "sigma": self.sigma,
-            "gamma": self.gamma,
-        }
-
-        return children, aux_data
+    random_key: KeyArrayLike
+    random_generator: _RandomGenerator
+    noise_conditioning: bool
+    use_analytic: bool
+    num_random_vectors: int
+    learning_rate: float
+    num_epochs: int
+    batch_size: int
+    hidden_dims: Sequence
+    optimiser: _LearningRateOptimiser
+    num_noise_models: int
+    sigma: float
+    gamma: float
 
     # pylint: disable=too-many-arguments
     def __init__(  # noqa: PLR0913, PLR0917
         self,
-        random_key: coreax.util.KeyArrayLike,
-        random_generator: Callable,
+        random_key: KeyArrayLike,
+        random_generator: _RandomGenerator,
         noise_conditioning: bool = True,
         use_analytic: bool = False,
         num_random_vectors: int = 1,
         learning_rate: float = 1e-3,
         num_epochs: int = 10,
         batch_size: int = 64,
-        hidden_dims: Optional[Sequence] = None,
-        optimiser: Callable = optax.adamw,
+        hidden_dims: Sequence = (128, 128, 128),
```

Comment:
```suggestion
        hidden_dims: Sequence[int] = (128, 128, 128),
```
and the same change for the declaration above.

Reply:
Added !